### ISSUE TITLE
390 raids stay at "1" completion

### ISSUE THREAD

**Issue by darkelement1987** [TrackerNetwork/DestinyStatus#66]
Title says all i guess, i did VoG/Crota 2 times on each character but counter stays at "1"

**Comment by iBotPeaches** [comment]
Interesting. I only did each 390 Raid once. I will attempt to replicate this, because we simply defer to the count from API, if that is returning 1 despite me doing another raid there is little I can do short of reporting a bug to Bungie.

**Comment by darkelement1987** [comment]
On destinychecklist.net it does count the way it should, doubt it is a API thing.

Example: https://www.destinychecklist.net/tabs/2/darkelement1987

I did crota 390:

Warlock 400: 3 times
Warlock 399: 3 times
Titan 398: 2 times

Still the counter on destinystatus is "1"

**Comment by iBotPeaches** [comment]
So did some digging. 390 Crota's End is `4000873610` in the API. So I debugged awaiting API return for all 3 of your characters

<img width="406" alt="screen shot 2017-04-16 at 7 53 31 am" src="https://cloud.githubusercontent.com/assets/611784/25070987/f1d935e8-2279-11e7-894b-6ad8b680dbfc.png">
<img width="400" alt="screen shot 2017-04-16 at 7 53 55 am" src="https://cloud.githubusercontent.com/assets/611784/25070988/f2fad238-2279-11e7-925d-c540a2127438.png">
<img width="413" alt="screen shot 2017-04-16 at 7 54 05 am" src="https://cloud.githubusercontent.com/assets/611784/25070989/f4639b64-2279-11e7-9582-a2d71fcd6006.png">

Every single API response from Bungie is 1. We use the aggregate endpoint. We don't crawl activity history looking for completions as that becomes impossible for accounts who have over 1k hours. DestinyChecklist isn't open source - I don't know what they do. 

I know the endpoint I rely on from Bungie, is not returning proper data.

**Comment by randalla0622** [comment]
Destinystatus.com is only showing completions for the 390LL featured version of each raid, and not the 390LL non-featured raids. For example, compare the VoG completions on my PSN account on destinystatus.com and fireteamops.com:

DS.com: http://destinystatus.com/psn/xaren0622
FO.com: http://fireteamops.com/psn/Xaren0622/raid-completions

FO.com has the normal, hard, 390LL and 390LL featured all split out. The numbers on DS.com for the 390LL VoG match the numbers from the 390LL featured on FO.com.

FO.com does not seem to show the non-completed 390LL versions of the raids. I've not done the 390LL non-featured WotM raid yet on any character, but have completed the 390LL featured WotM raid on one character. Maybe this is why you are not seeing that information in your API end-point?

**Comment by randalla0622** [comment]
Destinystatus.com is also not showing completions for the featured WotM raid.

**Comment by randalla0622** [comment]
So, for WotM, Bungie updated the existing 380LL heroic version to 390LL. This means that there is not a normal, heroic, 390LL and 390LL-featured version of the raid, just normal, 390LL and 390LL-featured. The other raids have their normal, hard/heroic, 390LL and 390LL-featured versions.

**Comment by iBotPeaches** [comment]
As far as my research goes, there is no 390LL WoTM. The Hard Mode WoTM is the same thing as the 390, upgraded per se. You can easily make that difference if WoTM is the featured by checking in the /advisor endpoint if the weekly raid was completed in the time of the featured raid when its WoTM.

However creating an aggregate count of the featured 390LL vs Heroic doesn't seem possible yet. My guess is that when WoTM is no longer the featured, FireTeamOps will have no way to determine it. 

I am still trying to find the reason the API isn't returning proper data for 390 Crota. I've asked a few other API providers. I am currently unsure why things aren't working.

**Comment by randalla0622** [comment]
Not saying you should, but you might ping dude@fireteamops.com and see if he has any suggestions.

**Comment by iBotPeaches** [comment]
Thanks. Problem identified. 390LL raids featured have different IDs than non featured 390LLs.

Unrelated, site is getting blasted dead. Working on uptime, then will get a patch out for this.

**Comment by iBotPeaches** [comment]
```
Kingsfall - 3978884648 (featured) - 1016659723 (Non featured)
Crota - 4000873610 (featured) - 2324706853 (non featured)
VOG - 856898338 (featured) - 4038697181 (non featured)
WOTM - 3356249023 (featured) - 430160982 (non featured)
```

Kudos to xgerhard for IDs. Will patch when I get home.

**Comment by iBotPeaches** [comment]
Okay, a hotfix is in. It isn't pretty, but it'll do the job for now. As for the WoTM 390LL vs Heroic. This is the situation.

The "Raids" part of the API where we pull Crota, VoG, Kings, etc all have their respective 3 activity tiers being Normal, Hard, 390.

WoTM on the other hand has two in that - Normal/Hard. However, since it's the featured raid, the data included in the featured index has this "tier 3" which is 390LL which is actually the Heroic raid. It seems they upgraded the Heroic from 380 to 390. You can test this by going to your director and trying to find 3 difficulties to select. You will only find two, so we have no easy way to make a checklist entry for 390LL yet.

**Comment by randalla0622** [comment]
I can verify that I see the changes as well. Hope you are able to find a solution for the WotM 390LL featured issue. I've fully completed two runs of the featured raid, and Aksis, phase 2 on my third character. None of those show on destinystatus.com, sadly.

**Comment by iBotPeaches** [comment]
![randalla](https://cloud.githubusercontent.com/assets/611784/25228458/88dadbae-259a-11e7-9f64-fa9c25c60080.png)
Titan, Hunter, Lock in terms of the columns.

Just to confirm this. You have completed 390LL WoTM on all characters once (completed = killing boss), but on your Warlock you only killed the boss. (I don't see checkpoint clears for the rest of the raid, only Aksis which is why there is no checkmark)

I unforunately haven't been able to play this week yet, so I don't have my own knowledge to cross reference. If this seems right, I think we are good.

**Comment by dcaslin** [comment]
@iBotPeaches Looks like you have this solved, DestinyChecklist.net is open source, I've just apparently been neglectful about linking to it anywhere =) 

Here it is: https://bitbucket.org/dcaslin/destinychecklist.node

TL;DR Summing the 390 Feature raid completions with the 390 non-featured raid completions is how I deal with this.

**Comment by iBotPeaches** [comment]
Thanks @dcaslin , guess I didn't look hard enough :/ 

I am summing them now as well. I will look into doing that for Arena (PoE 32/34/35 and Challenge of Elders) as well.

**Comment by darkelement1987** [comment]
Thanks for fixing!
